pages/architecture.md

```diff
@@ -122,15 +122,43 @@ the server hosting the documentation.
 
 ### VPC facing architecture
 In this architecture, data.all static sites are deployed on an AWS internal application load
-balancer (ALB) deployed on the VPC's private subnet. 
-This ALB is reachable only from Amazon VPCs and not from the internet. 
-Also, APIs are private and accessible only through VPC endpoints.
+balancer (ALB) deployed on the VPC's private subnet. Data.all static sites are hosted on Amazon ECS using docker containers through nginx server.
 
 
-Finally, data.all static sites are hosted on Amazon ECS using docker containers through nginx server.
+The ALB is reachable only from Amazon VPCs and not from the internet. Also, APIs are private and accessible only through VPC endpoints. 
+For this kind of architecture, the following resources need to be provisioned as pre-requisite for the deployment:
+- Route 53 private hosted zone
+- ACM certificate
+- For the above you will also need a VPC which needs to be provided as input for the deployment. Check the backend VPC section to review the VPC requirements.
 
+With the following commands you can create the ACM certificate and Route 53 private hosted zone:
+1.	`cd` to empty directory
+2.	This command will create your pem and a paraphrase password file: `openssl req -x509 -newkey rsa:4096 -days 1825 -keyout dataallkey.pem -out dataall.pem`
+3.	This command will create a no password file to load in ACM: `openssl rsa -in dataallkey.pem -out dataallkeynopwd.pem `
+4.	`aws route53 create-hosted-zone --name <domain-name> --vpc VPCRegion=<vpc_region>,VPCId=<vpc-id> --caller-reference 07:12:22 --query HostedZone.Id --output text `
+5.	`aws acm import-certificate --region us-east-1 --certificate fileb://<filepath to cert> --private-key fileb://<filepath to no password key> --query CertificateArn --output text`
 
 
+#### VPC facing + corporate network
+As stated above, the application in this case is accessible only from within the VPC. However, some customers require the ability to 
+connect from outside the VPC but within the corporate network. 
+
+As part of the deployment, data.all creates an API gateway and accesses it with the URL `https://<gateway-id>.execute-api.<region>.amazonaws.com`
+This works when we access the application from the VPC; but when we are outside of the VPC, the execute-api VPC endpoint needs to be added to the URL. 
+Hence the URL should be: `https://<gateway-id>.<execute-api-vpc-endpoint-id>.execute-api.<region>.amazonaws.com`.
+For this requirement we need to modify the code in `deploy/stacks/lambda_api.py` and adjust the api URL accordingly. 
```

Comment:
I wonder if we should consider making it configurable (maybe create an issue for it)?